sidekick_core/lib/src/sidekick_version_checker.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:convert';
+
+import 'package:deep_pick/deep_pick.dart';
+import 'package:http/http.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:yaml/yaml.dart';
+
+/// Checks and updates dependencies of the generated sidekick CLI
+class SidekickVersionChecker {
+  const SidekickVersionChecker();
+
+  /// Checks whether the latest version of [package] is being used in the generated sidekick CLI
+  ///
+  /// [pubspecPath] can be passed to override the default behavior of reading
+  /// the current package version from the pubspec.yaml at ['dependencies'][package]
+  /// E.g. ['dev_dependencies', 'some_dev_dependency'] or ['sidekick', 'generator_version']
+  Future<bool> isUpToDate({
+    required String package,
+    List<String>? pubspecPath,
+  }) async {
+    final latest = await getLatestPackageVersion(package);
+    final current = getCurrentMinimumPackageVersion(
+      pubspecPath ?? ['dependencies', package],
+    );
+
+    return latest == current;
+  }
+
+  /// Updates the version constraint of [package] to the latest version
+  Future<void> updateVersionConstraintToLatest(String package) async =>
+      updateVersionConstraint(
+        package: package,
+        newMinimumVersion: await getLatestPackageVersion(package),
+      );
+
+  /// Replaces the current version constraint of [package] in pubspec.yaml with [newMinimumVersion]
+  ///
+  /// If [pinVersion] is true, the new version constraint will only allow [newMinimumVersion]
+  /// Else it will allow a version range from [newMinimumVersion] until the next breaking version
+  void updateVersionConstraint({
+    required String package,
+    required Version newMinimumVersion,
+    bool pinVersion = false,
+  }) {
+    final pubspec = Repository.requiredSidekickPackage.pubspec;
+    final lines = pubspec.readAsLinesSync();
+
+    final newVersionConstraint = pinVersion
+        ? newMinimumVersion.canonicalizedVersion
+        : newMinimumVersion.major > 0
+            ? '^${newMinimumVersion.canonicalizedVersion}'
+            : "'>=${newMinimumVersion.canonicalizedVersion} <1.0.0'";
+
+    final index = lines.indexWhere((it) => it.startsWith('  $package:'));
+    lines[index] = '  $package: $newVersionConstraint';
+
+    pubspec.writeAsStringSync(lines.join('\n'));
```

Comment:
Ah that gave me the idea to use `dart pub remove <package> && dart pub add <package>:<version-constraint>` instead :)

Reply:
Nevermind I'll leave it as is because otherwise I can't reuse this method to update `sidekick: cli_version` in the pubspec.yaml since that isn't a dependency